### PR TITLE
add ESP8266WebServer::sendContent_P and ESP8266WebServer::send_P with contentLength

### PR DESCRIPTION
**Pull request by martinayotte**
- add ESP8266WebServer::sendContent_P and ESP8266WebServer::send_P with size_t contentLength argument to allow binary content to be send, such as images.
- fix in WiFiClient::write_P to use memcpy instead of memccpy allow binary content to be send, such as images.

### CODE REVIEW

File: hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer.cpp [hunk 1]
```diff
@@ -169,6 +169,13 @@ void ESP8266WebServer::send_P(int code, PGM_P content_type, PGM_P content) {
     sendContent_P(content);
 }
 
+void ESP8266WebServer::send_P(int code, PGM_P content_type, PGM_P content, size_t contentLength) {
+    String header;
+    _prepareHeader(header, code, String(FPSTR(content_type)).c_str(), contentLength);
```
Comment (igrr) [inline hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer.cpp]: Ouch, you are calling `c_str()` method of a temporary String instance here. Better move String declaration into a separate line.

**Comment by martinayotte** [comment]
Hi IGRR, 
I've simply copied the code from Makuna, line #167, so if it need to be changed, it need to be done on both.

**Comment by igrr** [comment]
Oh okay, somehow i missed that. I'll merge everything then and do the necessary changes.

**Comment by martinayotte** [comment]
Hi again Ivan,
I hate putting my hands/arms into a wringer.
during my previous tests, I forgot to add PSTR("image/png") in my calls.
Adding that, I got some crashes, even in Makuna original code.
So, I conclude that original code isn't working : 

prepareHeader(header, code, String(FPSTR(content_type)).c_str(), contentLength);

Changing that to what you suggest me doesn't work either : 

String type = String(FPSTR(content_type));
_prepareHeader(header, code, type.c_str(), contentLength);

So, I've think the proper way is again to use memccpy_P(), but I feel it is a bit ugly : 

char type[64];
memccpy_P((void*)type, (PGM_VOID_P)content_type, 0, sizeof(type));
_prepareHeader(header, code, (const char* )type, contentLength);

I will commit that for now, but I think it needs to be revisited...
I will appreciate you help here !

**Pull request merged**